multi_company_remove/models/res_company.py

```diff
@@ -141,24 +141,35 @@ def move_users(self, remaining_company):
             SET company_id = %s
             WHERE company_id IN %s
         """
-        cr.execute(statement, (remaining_company.id, self.ids))
-        user_model = self.env['res.users']
-        users_to_change = user_model.search([
-            ('company_id', 'in', self.ids),
-        ])
-        if users_to_change:
-            for user in users_to_change:
-                if remaining_company.id not in user.company_ids.ids:
```

Comment:
I do not see the SQL replacement for updating the allowed companies for the user. Note there are two fields on res_users: company_id and company_ids (plural).